### PR TITLE
Add documentation fro buildpack build time environment variables

### PR DESCRIPTION
**Pull request by JanakaSandaruwan**
## Purpose
https://github.com/wso2-enterprise/choreo/issues/28643

### CODE REVIEW

File: en/docs/develop-components/deploy-an-application-with-buildpacks.md [hunk 1]
```diff
@@ -249,4 +249,13 @@ Follow the guidelines below based on your language:
     e.g.,
 
     `web: java -jar target/sample.war` <br>
-`
+
+### Build Time Environment Variables
+
+You can configure the environment variables that are necessary to build the component using the Build Configurations Editor in the Build Tab. This is an optional option and not available for Ballerina and WSO2 MI.
```
Comment (Mirage20) [inline en/docs/develop-components/deploy-an-application-with-buildpacks.md:255]:
```suggestion
You can configure the environment variables that are necessary to build the component using the Build Configurations Editor in the component Build page. This option is not available for Ballerina and WSO2 MI.
```

File: en/docs/develop-components/deploy-an-application-with-buildpacks.md [hunk 2]
```diff
@@ -249,4 +249,13 @@ Follow the guidelines below based on your language:
...
+
+These environment variables and values will be passed to buildpack during the build time. Hence, this option can be used to configure buildpack specific environment variables or enviromemt variables required for component build.
+
+e.g.,
+
+- You can override the Maven command of `Java` buildpack using `GOOGLE_MAVEN_BUILD_ARGS` as environment key and `clean install` as value. Refer the following [documentation](https://cloud.google.com/docs/buildpacks/service-specific-configs) for more usages
```
Comment (Mirage20) [inline en/docs/develop-components/deploy-an-application-with-buildpacks.md:261]:
```suggestion
- You can override the Maven command of `Java` buildpack using `GOOGLE_MAVEN_BUILD_ARGS` as environment key and `clean install` as value. Refer the [documentation](https://cloud.google.com/docs/buildpacks/service-specific-configs) for more usages
```

Comment (Mirage20) [inline en/docs/develop-components/deploy-an-application-with-buildpacks.md:261]:
```suggestion
For example, you can override the Maven command of `Java` buildpack using `GOOGLE_MAVEN_BUILD_ARGS` as the environment key and `clean install` as the value. 

Refer to the [documentation](https://cloud.google.com/docs/buildpacks/service-specific-configs) for more usages.
```

I think we can separate the external documentation like this

Comment (JanakaSandaruwan) [reply en/docs/develop-components/deploy-an-application-with-buildpacks.md]: Fixed with https://github.com/wso2/docs-choreo-dev/pull/1506/commits/276829871a1bc1bc09226a8d614bf01e15e70d83

Comment (JanakaSandaruwan) [reply en/docs/develop-components/deploy-an-application-with-buildpacks.md]: Fixed with https://github.com/wso2/docs-choreo-dev/pull/1506/commits/276829871a1bc1bc09226a8d614bf01e15e70d83

File: en/docs/develop-components/deploy-an-application-with-buildpacks.md (hunk 1 above)
Comment (JanakaSandaruwan) [reply en/docs/develop-components/deploy-an-application-with-buildpacks.md]: Fixed with https://github.com/wso2/docs-choreo-dev/pull/1506/commits/276829871a1bc1bc09226a8d614bf01e15e70d83

**Review by Mirage20 — APPROVED** [review]

File: en/docs/develop-components/deploy-an-application-with-buildpacks.md [hunk 3]
```diff
@@ -249,4 +249,13 @@ Follow the guidelines below based on your language:
     e.g.,
 
     `web: java -jar target/sample.war` <br>
-`
+
+### Build Time Environment Variables
```
Comment (Nashaath) [inline en/docs/develop-components/deploy-an-application-with-buildpacks.md:253]:
```suggestion
## Configure build-time environment variables
```

File: en/docs/develop-components/deploy-an-application-with-buildpacks.md [hunk 4]
```diff
@@ -249,4 +249,13 @@ Follow the guidelines below based on your language:
...
+
+You can configure the environment variables that are necessary to build the component using the Build Configurations Editor in the component Build page. This option is not available for Ballerina and WSO2 MI.
```
Comment (Nashaath) [inline en/docs/develop-components/deploy-an-application-with-buildpacks.md:255]:
```suggestion
You can configure the environment variables necessary to build the component using the **Build Configurations Editor** on the component **Build** page. 

!!! info
    The capability to configure build-time environment variables is not available for components created using **Ballerina** or **WSO2 MI** buildpacks.
```

File: en/docs/develop-components/deploy-an-application-with-buildpacks.md [hunk 5]
```diff
@@ -249,4 +249,13 @@ Follow the guidelines below based on your language:
...
+
+These environment variables and values will be passed to buildpack during the build time. Hence, this option can be used to configure buildpack specific environment variables or enviromemt variables required for component build.
```
Comment (Nashaath) [inline en/docs/develop-components/deploy-an-application-with-buildpacks.md:257]: During the build process, the build-time environment variables and their values are passed to the buildpack. Therefore, you can configure both buildpack-specific environment variables and those required for the component build.

File: en/docs/develop-components/deploy-an-application-with-buildpacks.md [hunk 6]
```diff
@@ -249,4 +249,13 @@ Follow the guidelines below based on your language:
...
+
+For example, you can override the Maven command of `Java` buildpack using `GOOGLE_MAVEN_BUILD_ARGS` as the environment key and `clean install` as the value.
```
Comment (Nashaath) [inline en/docs/develop-components/deploy-an-application-with-buildpacks.md:259]: For example, if you want to override the Maven command of the **Java** buildpack, you can use `GOOGLE_MAVEN_BUILD_ARGS` as the environment key and `clean install` as the value.

Comment (Nashaath) [inline en/docs/develop-components/deploy-an-application-with-buildpacks.md:259]:
```suggestion
For example, if you want to override the Maven command of the **Java** buildpack, you can use `GOOGLE_MAVEN_BUILD_ARGS` as the environment key and `clean install` as the value. 
```

File: en/docs/develop-components/deploy-an-application-with-buildpacks.md (hunk 5 above)
Comment (Nashaath) [inline en/docs/develop-components/deploy-an-application-with-buildpacks.md:257]:
```suggestion
During the build process, the build-time environment variables and their values are passed to the buildpack. Therefore, you can configure both buildpack-specific environment variables and those required for the component build.
```

File: en/docs/develop-components/deploy-an-application-with-buildpacks.md [hunk 7]
```diff
@@ -249,4 +249,13 @@ Follow the guidelines below based on your language:
...
+
+### Build Time Environment Variables
+
+You can configure the environment variables that are necessary to build the component using the Build Configurations Editor in the component Build page. This option is not available for Ballerina and WSO2 MI.
+
+These environment variables and values will be passed to buildpack during the build time. Hence, this option can be used to configure buildpack specific environment variables or enviromemt variables required for component build.
+
+For example, you can override the Maven command of `Java` buildpack using `GOOGLE_MAVEN_BUILD_ARGS` as the environment key and `clean install` as the value. 
+
+Refer to the [documentation](https://cloud.google.com/docs/buildpacks/service-specific-configs) for more usages.
```
Comment (Nashaath) [inline en/docs/develop-components/deploy-an-application-with-buildpacks.md:261]:
```suggestion
For more examples, see [Google Cloud's buildpacks documentation](https://cloud.google.com/docs/buildpacks/service-specific-configs).
```

**Review by Nashaath — COMMENTED** [review]
Added a few minor suggestions to apply if applicable.

File: en/docs/develop-components/deploy-an-application-with-buildpacks.md (hunk 6 above)
Comment (JanakaSandaruwan) [reply en/docs/develop-components/deploy-an-application-with-buildpacks.md]: Fixed with https://github.com/wso2/docs-choreo-dev/pull/1506/commits/abd0d07a2644090eb9f1d9f06bf24e9d028eb409

File: en/docs/develop-components/deploy-an-application-with-buildpacks.md (hunk 3 above)
Comment (JanakaSandaruwan) [reply en/docs/develop-components/deploy-an-application-with-buildpacks.md]: Fixed with https://github.com/wso2/docs-choreo-dev/pull/1506/commits/efb3937a9eac353fbde7d4f1a1c7f007848704f3

File: en/docs/develop-components/deploy-an-application-with-buildpacks.md (hunk 4 above)
Comment (JanakaSandaruwan) [reply en/docs/develop-components/deploy-an-application-with-buildpacks.md]: Fixed with https://github.com/wso2/docs-choreo-dev/pull/1506/commits/efb3937a9eac353fbde7d4f1a1c7f007848704f3

File: en/docs/develop-components/deploy-an-application-with-buildpacks.md (hunk 5 above)
Comment (JanakaSandaruwan) [reply en/docs/develop-components/deploy-an-application-with-buildpacks.md]: Fixed with https://github.com/wso2/docs-choreo-dev/pull/1506/commits/efb3937a9eac353fbde7d4f1a1c7f007848704f3

File: en/docs/develop-components/deploy-an-application-with-buildpacks.md (hunk 7 above)
Comment (JanakaSandaruwan) [reply en/docs/develop-components/deploy-an-application-with-buildpacks.md]: Fixed with https://github.com/wso2/docs-choreo-dev/pull/1506/commits/efb3937a9eac353fbde7d4f1a1c7f007848704f3

**Review by Nashaath — APPROVED** [review]

**Pull request merged**